packages/php-db-import-export/tests/functional/Snowflake/ToFinal/SqlBuilderTest.php

```diff
@@ -1621,6 +1622,358 @@ public function testGetUpdateWithPkCommandNullManipulationWithTimestampFeatures(
         ], $result);
     }
 
+    public function testGetUpdateWithPkCommandNullManipulationSpecialDatatypes(): void
+    {
+        $timestampInit = new DateTime('2020-01-01 00:00:01');
+        $timestampSet = new DateTime('2020-01-01 01:01:01');
+        $this->createTestSchema();
+        $tableDefinition = new SnowflakeTableDefinition(
+            self::TEST_SCHEMA,
+            self::TEST_TABLE,
+            false,
+            new ColumnCollection([
+                new SnowflakeColumn(
+                    'id',
+                    new Snowflake(
+                        Snowflake::TYPE_INT,
+                    ),
+                ),
+                new SnowflakeColumn(
+                    'array',
+                    new Snowflake(
+                        Snowflake::TYPE_ARRAY,
+                        ['nullable' => true],
+                    ),
+                ),
+                new SnowflakeColumn(
+                    'geometry',
+                    new Snowflake(
+                        Snowflake::TYPE_GEOMETRY,
+                        ['nullable' => true],
+                    ),
+                ),
+                new SnowflakeColumn(
+                    'geography',
+                    new Snowflake(
+                        Snowflake::TYPE_GEOGRAPHY,
+                        ['nullable' => true],
+                    ),
+                ),
+                new SnowflakeColumn(
+                    '_timestamp',
+                    new Snowflake(Snowflake::TYPE_TIMESTAMP),
+                ),
+            ]),
+            ['id'],
+        );
+        $stageDefinition = new SnowflakeTableDefinition(
+            self::TEST_SCHEMA,
+            self::TEST_STAGING_TABLE,
+            false,
+            new ColumnCollection([
+                new SnowflakeColumn(
+                    'id',
+                    new Snowflake(
+                        Snowflake::TYPE_INT,
+                    ),
+                ),
+                new SnowflakeColumn(
+                    'array',
+                    new Snowflake(
+                        Snowflake::TYPE_ARRAY,
+                        ['nullable' => true],
+                    ),
+                ),
+                new SnowflakeColumn(
+                    'geometry',
+                    new Snowflake(
+                        Snowflake::TYPE_GEOMETRY,
+                        ['nullable' => true],
+                    ),
+                ),
+                new SnowflakeColumn(
+                    'geography',
+                    new Snowflake(
+                        Snowflake::TYPE_GEOGRAPHY,
+                        ['nullable' => true],
+                    ),
+                ),
+            ]),
```

Comment:
Pozeral som a vsade pouzivame nejake genericke stlpce pre stagin tabulku napr pouzivame na to `createNullableGenericColumn`, takze staging tabulka je stale varchar nie? Potom tento test je trochu zavadzajuci ze ma typy.

Reply:
To sa právě změní https://github.com/keboola/connection/pull/5181 popisuju to v tom jira ticketu

Reply:
Jo to som zaznamenal, zesa to ma menit akurat som nevedel ze uz tu, tak asi ok ;)